lib/src/core/pointcut.ts

```diff
@@ -6,7 +6,14 @@ export class Pointcut {
 
   public advice: Advice;
 
+  private _applications = new Set<Function>();
+
   public apply(fn: Function, woveMetadata: any) {
+    if (this._applications.has(fn)) {
+      return;
+    } else {
```

Comment:
You could remove the else because you return in the if. Depends on code style :).